instrumentation/active_record/lib/opentelemetry/instrumentation/active_record/patches/validations.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActiveRecord
+      module Patches
+        # Module to prepend to ActiveRecord::Validations for instrumentation
+        # We patch these methods because they either raise, or call
+        # the super implementation in persistence.rb
+        # https://github.com/rails/rails/blob/v5.2.4.5/activerecord/lib/active_record/validations.rb#L42-L53
```

Comment:
I moved the patching of `save` and `save!` from the persistence class to the validations class , as we won't trace the call to either if it fails a validation at this level.